### PR TITLE
Integration tests executed on a real deployment as part of the CICD - Redshift Shares

### PR DESCRIPTION
**Pull request by dlpzx**
⚠️ MERGE AFTER #1636 
### Feature or Bugfix
- Feature: Testing

### Detail
Add integration tests for Redshift shares. Implements #1620 
- Implemented inside the shares modules in a subdirectory so that each share type can have its own conftest but still re-use common methods from shares queries
- This PR is focused on testing the Redshift shares functionality, it does not include all tests that test the workflow of the share (e.g. submit, reject...)
- It does not validate if after a share the user has access to data. We could implement it using the Redshift Data API, but I left it as optional for a separate PR

### Tested
Locally:
image

### Relates
- https://github.com/data-dot-all/dataall/issues/1620
- https://github.com/data-dot-all/dataall/issues/1619
- https://github.com/data-dot-all/dataall/issues/1220

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization?
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features?
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests_new/integration_tests/modules/redshift_datasets/global_conftest.py
Comment (petrkalos) [inline tests_new/integration_tests/modules/redshift_datasets/global_conftest.py:1]: would you mind applying [this commit](https://github.com/data-dot-all/dataall/pull/1649/commits/85969b3d053121245194c050a4774f4b59e32dc8) to make the redshift tests optional and also implement the same for the `redshit_datasets_shares`?

File: tests_new/integration_tests/modules/shares/redshift_datasets_shares/aws_clients.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    def deauthorize_datashare(self, datashare_arn: str, target_account: str) -> Dict[str, Any]:
+        log.info('Deauthorizing Redshift datashare...')
+        try:
+            response = self._client.deauthorize_data_share(
+                DataShareArn=datashare_arn, ConsumerIdentifier=target_account
+            )
+            log.info(f'Datashare deauthorized successfully: {datashare_arn}')
+            return response
+        except ClientError as e:
+            log.error(f'Error deauthorizing datashare: {e}')
```
Comment (petrkalos) [inline tests_new/integration_tests/modules/shares/redshift_datasets_shares/aws_clients.py:22]: nit: log.exception

**Review by petrkalos — APPROVED** [review]
lgtm, feel free to merge after incorporating the "optional" tests commit

File: tests_new/integration_tests/modules/redshift_datasets/global_conftest.py
Comment (dlpzx) [reply tests_new/integration_tests/modules/redshift_datasets/global_conftest.py:1]: Added! Yes, it makes more sense here. The pytest skip is ultra cool

**Pull request merged**